contracts/clients-tests/BlsProvider.test.ts

```diff
@@ -0,0 +1,535 @@
+/* eslint-disable camelcase */
+import { ethers as hardhatEthers } from "hardhat";
+import chai, { expect } from "chai";
+import spies from "chai-spies";
+import { ethers, Wallet } from "ethers";
+import { parseEther, formatEther, id } from "ethers/lib/utils";
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+
+import {
+  BlsProvider,
+  BlsSigner,
+  BlsWalletWrapper,
+  MockERC20__factory,
+  NetworkConfig,
+} from "../clients/src";
+import getNetworkConfig from "../shared/helpers/getNetworkConfig";
+
+chai.use(spies);
+
+let networkConfig: NetworkConfig;
+let signers: SignerWithAddress[];
+
+let aggregatorUrl: string;
+let verificationGateway: string;
+let rpcUrl: string;
+let network: ethers.providers.Networkish;
+
+let privateKey: string;
+let blsProvider: BlsProvider;
+let blsSigner: BlsSigner;
+
+let regularProvider: ethers.providers.JsonRpcProvider;
+let regularSigner: ethers.providers.JsonRpcSigner;
+
+describe("BlsProvider", () => {
+  beforeEach(async () => {
+    networkConfig = await getNetworkConfig("local");
+    signers = await hardhatEthers.getSigners();
+
+    aggregatorUrl = "http://localhost:3000";
+    verificationGateway = networkConfig.addresses.verificationGateway;
+    rpcUrl = "http://localhost:8545";
+    network = {
+      name: "localhost",
+      chainId: 0x7a69,
+    };
+    privateKey = Wallet.createRandom().privateKey;
+
+    regularProvider = new ethers.providers.JsonRpcProvider(rpcUrl);
+    regularSigner = regularProvider.getSigner();
+
+    blsProvider = new BlsProvider(
+      aggregatorUrl,
+      verificationGateway,
+      rpcUrl,
+      network,
+    );
+    blsSigner = blsProvider.getSigner();
+    await blsSigner.initWallet(privateKey);
+
+    const fundedWallet = new ethers.Wallet(
+      "0x59c6995e998f97a5a0044966f0945389dc9e86dae88c7a8412f4603b6b78690d",
+      regularProvider,
+    );
+
+    await fundedWallet.sendTransaction({
+      to: blsSigner.wallet.address,
+      value: parseEther("1"),
+    });
+  });
+
+  it("should return a valid signer", async () => {
+    // Arrange & Act
+    const blsSigner = blsProvider.getSigner();
+
+    // Assert
+    /* eslint-disable-next-line no-unused-expressions */
+    expect(blsSigner._isSigner).to.be.true;
```

Comment:
I would just disable this rule in test files. See https://stackoverflow.com/a/37777342

```suggestion
    expect(blsSigner._isSigner).to.be.true;
```

Reply:
Much nicer, thanks